File: java/src/main/java/ch/planval/datavisualizer/model/Source.java
package ch.planval.datavisualizer.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.util.List;
import java.util.UUID;

@Data
@Table(name = "sources")
@Entity
@AllArgsConstructor
@RequiredArgsConstructor
public class Source {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid")
    private UUID id;
    @Column(name = "url")
    private String url;
    @Column(name = "api_url")
    private String api_url;
    @Column(name = "title")
    private String title;
}

File: ui/sustainability-ui/src/app/models/source.ts
export class Source {
    id: string;
    url: string;
    api_url: string;
    title: string;

    constructor(id: string, url: string, api_url: string, title: string) {
        this.id = id;
        this.url = url;
        this.api_url = api_url;
        this.title = title;
    }
}
